### PR TITLE
feat: automate uploads for maestro copilot builds

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [PHIRE-850] 

### Description

Automate uploads for maestro builds via a script, can be moved to be a ci job but since we don't want to trigger it yet maybe this is enough?

Happy to hear your thoughts.

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- automate uploads for maestro copilot builds - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-850]: https://artsyproduct.atlassian.net/browse/PHIRE-850?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: scripts/maestro/build-android.sh [hunk 1]
```diff
@@ -21,6 +21,17 @@ fi
...
+
+# Check if the directory exists if not create it
+if [ ! -d "$directory" ]; then
+    echo "Directory does not exist. Creating..."
+    mkdir -p "$directory"
+    echo "Directory created successfully."
+else
+    echo "Directory already exists."
+fi
+
```
Comment (gkartalis) [inline scripts/maestro/build-android.sh:34]: creates `android/app/build/outputs/apk/release` dir if not there ~ encountered some cases where this was not there due to cleaning the project

File: scripts/maestro/build-ios.sh [hunk 2]
```diff
@@ -10,7 +10,17 @@ CI=1 yarn pod-install
 # Generate the iOS build file
 yarn maestro:ios:release:build
 
-# Open the folder where the build file is located
-open ios/build/Build/Products/Store-iphonesimulator
+# Navigate to the directory containing the build
+cd ios/build/Build/Products/Store-iphonesimulator
+
+# Zip the build directory
+zip -r Artsy.zip Artsy.app
```
Comment (gkartalis) [inline scripts/maestro/build-ios.sh:17]: zips the Artsy.app to be able to upload it later to the cloud

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 14525ef6a58ef4c4cccdd63eed055bdc2284924f
</p>

File: scripts/maestro/build-android.sh [hunk 3]
```diff
@@ -29,7 +40,13 @@ yarn maestro:android:release:build
 # correctly. You can ignore the error and continue with the script.
 yarn maestro:android:release:install
 
-# Open the folder where the build file is located
-open android/app/build/outputs/apk/release
+cd android/app/build/outputs/apk/release
+
+echo "Uploading the Android build to maestro server"
+
+curl https://api.copilot.mobile.dev/v2/project/$MAESTRO_ANDROID_PROJECT_ID/build \
```
Comment (brainbicycle) [inline scripts/maestro/build-android.sh:47]: do we have to set these in CI or export them somewhere?

**Review by brainbicycle — DISMISSED** [review]
this looks good, just wondering if we need a script of how we are getting the vars set in ci?

File: scripts/maestro/build-android.sh (hunk 3 above)
Comment (gkartalis) [reply scripts/maestro/build-android.sh:47]: I've already set them on CI, but not sure if we need extra actions in order to use them from there. Got inspired from [here](https://github.com/artsy/eigen/blob/41c145db05947dccc3877972f270d108c0609c69/scripts/utils/commit-hash-from-pr.sh#L9-L18) for this change but maybe we need to do something like [this](https://github.com/artsy/eigen/blob/41c145db05947dccc3877972f270d108c0609c69/scripts/utils/commit-hash-from-pr.sh#L9-L10)?

Comment (brainbicycle) [reply scripts/maestro/build-android.sh:47]: I think you might, maybe put in a test job and see what happens?

File: .circleci/config.yml [hunk 4]
```diff
@@ -427,6 +430,9 @@ jobs:
       - run:
           name: Save success file
           command: echo yes > 'build-success.log'
+      - run:
+          name: Upload builds to Maestro App Quality Copilot
+          command: ./scripts/maestro/build-android.sh
```
Comment (gkartalis) [inline .circleci/config.yml:435]:
```suggestion
```

**Pull request merged**